bitbots_docs/docs/manual/tutorials/lowlevel.rst

```diff
@@ -0,0 +1,168 @@
+These controllers are hardware agnostic because they are using interfaces to abstract from the hardware (e.g. motors).
+To control the motors, the Dynamixel Controller is used, which itself uses the Dynamixel Hardware Interface.
+
+ROS messages
+~~~~~~~~~~~~
+
+After all these steps we're finally at the ROS message level.
+There are two message types handled here: The joint state message gives the current positions of the motors, while the joint goals message can specify target positions for motors. The hardware interface also manages the IMU data and the values returned by the foot sensors.
+
+How do we use bitbots_ros_control?
+----------------------------------
+
+The package bitbots_ros_control provides the hardware interface for the dynamixel motors.
+
+The most important configuration file for this is the wolfgang.yaml. In this file you find multiple settings for defining, which values should be read from the motors (temperature, speed, force, ...), which sensors should be used (foot pressure sensors, IMU) and which settings should be set (control loop frequency, baud rate, CORE ports, auto torque, ...).
+
+The corresponding ROS node can be launched with `roslaunch bitbots_ros_control ros_control.launch`. This will execute the following steps:
+
+1. The motors are pinged in alphabetical order. This happens due to the way yaml files are read. This means the HeadPan motor (id 19) is read first, while the RShoulderRoll motor (id 3) is read last.
+2. The values from the config file are written into the RAM and ROM of the motors. These are values like speed or return delay time.
+3. The message "Hardware interface init finished" appears.
+4. The control loop starts, alternating between sync read and sync write.
+5. The controllers for ROS control are loaded.
+
+Help, I have a problem!
+-----------------------
+
+Error Opening Serial Port
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you encounter the message "Error opening serial port", no connection between the NUC and the CORE board could be established. Therefore your first instinct should be checking whether the cable is plugged in correctly. If this does not solve the problem, you can check whether the board can be found by using `lsusb` (look at the "leaf" entry). You can further investigate this by using `ls /dev/`. You should find the devices "/dev/ttyUSB0" through "/dev/ttyUSB3", one for each of the four busses. If the names are different, you may have to alter the wolfgang.yaml file or unplug the CORE board and plug it back in, in order to make it use the known names.
+
+Motor problems
+~~~~~~~~~~~~~~
+
+The first thing you should do if you have a motor problem ("no status from id", motors stuttering, ...) is checking whether all cables are plugged in correctly. Sometimes one of the cable sits loosely in its socket and may fall out entirely when the robot moves. To control whether all motors are reachable, this `software
```

Comment:
```suggestion
The first thing you should do if you have a motor problem ("no status from id", motors stuttering, ...) is checking whether all cables in the bus with the problem are plugged in correctly. Sometimes one of the cable sits loosely in its socket and may fall out entirely when the robot moves. To control whether all motors are reachable, this `software
```

Reply:
This edit should not be added, as even unrelated cables can cause issues in other motors. Of course it makes sense to check the affected cables first, but in general it makes sense to go through all cables, at least of that bus, to make sure.

Reply:
```suggestion
The first thing you should do if you have a motor problem ("no status from id", motors stuttering, ...) is checking whether all cables are plugged in correctly. Preferably starting with the bus with the motor that has a problem. Sometimes one of the cable sits loosely in its socket and may fall out entirely when the robot moves. To control whether all motors are reachable, this `software
```